### DISCUSSION TITLE
Improved HATEOAS+HAL Support

### DISCUSSION THREAD

**Discussion by SenorPez** [micronaut-projects/micronaut-core#9248] [General]
One of the biggest things I miss from Spring (it's offset by other positives, though!) is the mature support for HATEOAS link generation and workflow through controllers. Making my REST API HATEOAS+HAL compliant takes a bit more manual work in Micronaut.

Am I missing a plugin? Is there a project to improve support on this end that I might follow or even possibly contribute to?

**Comment by graemerocher** [comment]
there is not currently, but I am aware it is something we can improve on. The new API in Micronaut 4 for `MessageBodyReader` and `MessageBodyWriter` will help here

**Reply by SenorPez** [reply]
Really looking forward to that! Thanks!

**Reply by jaecktec** [reply]
I'm a bit confused how this API would help. Would it be possible to post a small example how I could use it?
